src/core/pal/feature.cpp

```diff
@@ -1005,25 +995,34 @@ LabelPosition *FeaturePart::curvedPlacementAtOffset( PointSet *path_positions, d
   {
     // Calculate the orientation based on the angle of the path segment under consideration
 
-    double _distance = distance;
+    double _distance = offsetAlongSegment;
     int endindex = index;
 
+    double startLabelX = 0;
+    double startLabelY = 0;
+    double endLabelX = 0;
+    double endLabelY = 0;
     for ( int i = 0; i < li->char_num; i++ )
     {
       LabelInfo::CharacterInfo &ci = li->char_info[i];
-      double start_x, start_y, end_x, end_y;
-      if ( !nextCharPosition( ci.width, path_distances[index], path_positions, endindex, _distance, start_x, start_y, end_x, end_y ) )
+      double start_x, start_y;
```

Comment:
Since you are cleaning up the underscores from variable names, should we use startX and startY instead of start_x and start_y here?